### ISSUE TITLE
OmniXAI install error

### ISSUE THREAD

**Issue by mChowdhury-91** [salesforce/OmniXAI#15]
While installing OmniXAI with pip install getting an error: "Building wheel for hnswlib (pyproject.toml) ... error
  error: subprocess-exited-with-error"

Here is the error log:
  Building wheel for hnswlib (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for hnswlib (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [57 lines of output]
      running bdist_wheel
      running build
      running build_ext
      creating tmp
      gcc -pthread -B /opt/conda/envs/omnixai/compiler_compat -Wno-unused-result -Wsign-compare -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/envs/omnixai/include -fPIC -O2 -isystem /opt/conda/envs/omnixai/include -fPIC -I/opt/conda/envs/omnixai/include/python3.10 -c /tmp/tmpy7q_qfxa.cpp -o tmp/tmpy7q_qfxa.o -std=c++14
      gcc -pthread -B /opt/conda/envs/omnixai/compiler_compat -Wno-unused-result -Wsign-compare -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/envs/omnixai/include -fPIC -O2 -isystem /opt/conda/envs/omnixai/include -fPIC -I/opt/conda/envs/omnixai/include/python3.10 -c /tmp/tmpdgx0wt9f.cpp -o tmp/tmpdgx0wt9f.o -std=c++11
      Traceback (most recent call last):
        File "/opt/conda/envs/omnixai/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
          main()
        File "/opt/conda/envs/omnixai/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/opt/conda/envs/omnixai/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 261, in build_wheel
          return _build_backend().build_wheel(wheel_directory, config_settings,
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 244, in build_wheel
          return self._build_with_temp_dir(['bdist_wheel'], '.whl',
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 229, in _build_with_temp_dir
          self.run_setup()
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 174, in run_setup
          exec(code, locals())
        File "<string>", line 116, in <module>
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/__init__.py", line 87, in setup
          return distutils.core.setup(**attrs)
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 177, in setup
          return run_commands(dist)
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 193, in run_commands
          dist.run_commands()
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 968, in run_commands
          self.run_command(cmd)
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 1217, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/wheel/bdist_wheel.py", line 299, in run
          self.run_command('build')
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 317, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 1217, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/command/build.py", line 24, in run
          super().run()
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/_distutils/command/build.py", line 131, in run
          self.run_command(cmd_name)
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 317, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 1217, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/command/build_ext.py", line 79, in run
          _build_ext.run(self)
        File "/tmp/pip-build-env-vkcm82xu/overlay/lib/python3.10/site-packages/setuptools/_distutils/command/build_ext.py", line 339, in run
          self.build_extensions()
        File "<string>", line 103, in build_extensions
        File "<string>", line 70, in cpp_flag
      RuntimeError: Unsupported compiler -- at least C++11 support is needed!
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for hnswlib
Failed to build hnswlib
ERROR: Could not build wheels for hnswlib, which is required to install pyproject.toml-based projects

**Comment by yangwenz** [comment]
No hesitate to contact us if there are still some problems in installing hnswlib. And could you list your OS version and python version?

**Comment by yangwenzhuo08** [comment]
Has the system installed "build-essential", e.g., apt install build-essential?

**Comment by MaxPolak97** [comment]
How did you solved this issue? I still running to the same problem and installing hnswlib doesn't work either as I'm getting the same error...

**Comment by yangwenz** [comment]
Hi, Max, if there is still some problem with hnswlib, please try to install
it separately as mentioned in https://github.com/nmslib/hnswlib. The
problem is related to CPU versions.

**Comment by arthurbabey** [comment]
Hi, I have a similar issue installing omnixai

I am using python 3.10 on windows 10. 

Here is the prompt: 

  error: subprocess-exited-with-error

  × Building wheel for hnswlib (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [5 lines of output]
      running bdist_wheel
      running build
      running build_ext
      building 'hnswlib' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for hnswlib
Failed to build hnswlib
ERROR: Could not build wheels for hnswlib, which is required to install pyproject.toml-based projects


I attempted to download Microsoft C++ Build Tools as instructed by the error prompt, and then I tried to install only hnswlib. However, the same error persists. Any assistance would be greatly appreciated.
